README.md

```diff
@@ -1,2 +1,202 @@
+- [`cleanupPromise()`](#cleanup-promise) - versatily promise variant that allows react-style hook cleanups
+- [`composeAbort()`](#compose-abort) - compose an abort-signal into sub-variants.
+- [`isPromiseLike()`](#is-promise-like) - typescript type-check helper
+- [`raceWithSignal()`](#race-with-signal) - races several promises, all but the first will be aborted
+- [`toPromise()`](#to-promise) - turns a value into a promise, cleverly
+- [`wrapTimeout()`](#wrap-timeout) - wraps an async function to have an optional timeout
+
+#### `bubbleAbort()`
+
+Simple function that throws an AbortError if an AbortSignal
+happens to be aborted.
+
+_Usage:_
+
+```javascript
+const { bubbleAbort } = require('@consento/promise/bubbleAbort')
+
+async function longRunning ({ signal } = {}) {
+  let result = ''
+  for await (const data of internalIterator()) {
+    bubbleAbort(signal) // Will throw an error if the signal is aborted
+    result += data
+  }
+  return result
+}
+```
+
+#### `cbPromise()`
+
+Extended Promise that has the `resolve` and `reject` methods
+on the class instance instead of a template function.
+
+```javascript
+const { CBPromise } = require('@consento/promise/cbPromise')
+
+const p = new CBPromise()
+p.cb(new Error(), 'result') // traditional callback API
+p.resolve('result') // or use traditional resolve
+p.reject(new Error()) // and reject
+```
+
+This may come in handy when you need a callback but want to await
+the promise.
+
+```javascript
+async function test () {
+  const p = new CBPromise()
+  setTimeout(p.resolve, 500)
+  await p
+}
+```
+
+#### `checkpoint()`
+
+Allows the creation of a checkpoint function that aborts
+an async script if a signal is aborted.
+
+_Usage:_
+
+```javascript
+const { checkpoint } = require('@consento/promise/checkpoint')
+
+async function longRunning ({ signal }: {}) {
+  const cp = checkpoint(signal)
+  let result
+  for (const data of internalIterator()) {
+    result += await cp(data) // An AbortError will be thrown if the passed-in signal happens to be aborted.
+  }
+  return result
+}
+```
+
+#### `cleanupPromise()`
+
+Versatile custom Promise implementation that allows to execute an async
+cleanup operation after a promise is resolved or rejected.
+
+_Usage:_
+
+```javascript
+import { cleanupPromise } = require('@consento/promise/cleanupPromise')
+
+const result = await cleanupPromise((resolve, reject, signal) => {
+  setTimeout(resolve, Math.random() * 500) // Resolve & reject work like in regular promises
+  setTimeout(reject, Math.random() * 500)
+  const abortHandler = () => {
+    reject(new Error('aborted'))
+  }
+  signal.addEventListener('abort', abortHandler)
+  return () => {
+    // Executed after resolve or reject is called.
+    signal.removeEventListener('abort, abortHandler)
+  }
+})
+
+cleanupPromise(..., { timeout: 500, signal }) // You can also pass-in a parent signal or a timeout!
+```
+
+#### `isPromiseLike()`
```

Comment:
Maybe we should have the Typescript interface here?

Also might be good to have an example of where you'd need this utility.

Reply:
added.